packages/angular_devkit/build_angular/src/angular-cli-files/utilities/check-port.ts

```diff
@@ -5,26 +5,34 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+const net = require('net');
 
-import { Observable } from 'rxjs';
-const portfinder = require('portfinder');
+export function checkPort(port: number, host: string, basePort = 49152): Promise<number> {
+  return new Promise<number>((resolve, reject) => {
+    function _getPort(portNumber: number) {
+      if (portNumber > 65535) {
+        reject(new Error(`There is no port available.`));
+      }
 
+      const server = net.createServer();
 
-export function checkPort(port: number, host: string, basePort = 49152): Observable<number> {
-  return new Observable(obs => {
-    portfinder.basePort = basePort;
-    // tslint:disable:no-any
-    portfinder.getPort({ port, host }, (err: any, foundPort: number) => {
-      if (err) {
-        obs.error(err);
-      } else if (port !== foundPort && port !== 0) {
-        // If the port isn't available and we weren't looking for any port, throw error.
-        obs.error(`Port ${port} is already in use. Use '--port' to specify a different port.`);
-      } else {
-        // Otherwise, our found port is good.
-        obs.next(foundPort);
-        obs.complete();
-      }
-    });
+      server.once('error', err => {
+        if (port !== 0 && err.code === 'EADDRINUSE') {
+          // If the port isn't available and we weren't looking for any port, throw error.
+          reject(`Port ${port} is already in use. Use '--port' to specify a different port.`);
```

Comment:
Shouldn't this be `new Error`?

Reply:
Exceeds the maximum line length. Any idea how can I do this?

Reply:
```
reject(
    new Error`Port ${port} is already in use. Use '--port' to specify a different port.`
);
```